### PR TITLE
Finish preprocessing migration

### PR DESCRIPTION
**Pull request by roshankern**
In this PR we:
- Finish the migration of the preprocessing module
- Lint/format code with Ruff
- Convert all docstrings to Google format

**Note**: I did not change the code to use the `nd2` package instead of `nd2reader` in this PR because it will require more testing.

@mat10d This is ready for review!

### CODE REVIEW

File: README.md
Comment (mat10d) [inline README.md:34]: This is very good. Let's keep this up!

File: workflow/lib/preprocess/preprocess.py
Comment (mat10d) [inline workflow/lib/preprocess/preprocess.py:82]: So as we migrate, are we assuming this format? Are we ignoring the potential to use well based images?

File: workflow/lib/shared/illumination_correction.py
Comment (mat10d) [inline workflow/lib/shared/illumination_correction.py:1]: Based on our discussion yesterday, I think you do have some logic to keep this file in shared. Might it make sense to have some sort of descriptor for each file? I think this was useful in the original codebase (main) that I had, but up to you. But this way, we could motivate the need to keep all the illumination correction functions in shared because it both calculates the correction and then applies it (you will add this later)

Comment (mat10d) [reply workflow/lib/shared/illumination_correction.py:1]: I don't know if there is a format, (e.g. one sentence per line/ruff/google docstrings) for doing this, or if it isn't conventional to have that. But worth a thought.

File: workflow/rules/preprocess.smk
Comment (mat10d) [inline workflow/rules/preprocess.smk:22]: I haven't seen this before, but get_sample_fps is a function that returns wildcards?

File: workflow/scripts/preprocess/nd2_to_tif.py
Comment (mat10d) [inline workflow/scripts/preprocess/nd2_to_tif.py:11]: I think before you merge this, it would be good to test if the things that you are saving are compatible with Fiji. I would download Fiji and Napari

https://imagej.net/software/fiji/downloads
https://napari.org/dev/tutorials/fundamentals/installation.html

And see what this tiff format is compatible with

We will also need to check if things work well with downstream steps given that we are massively changing how we are saving images. I am assuming that you can compare the analysis on the few tiles that you did in dev with these. I am a bit worried about this, but also understand the importance of streamlining this

File: workflow/lib/preprocess/file_utils.py
Comment (mat10d) [inline workflow/lib/preprocess/file_utils.py:7]: I really like this approach. I have struggled heavily with wildcard usage downstream of preprocessing. Let's keep this up. Will be really useful for merge/hash (step 3)

**Review by mat10d — APPROVED** [review]
I would check Napari and Fiji compatibility, just to know if the output tiffs are compatible. But otherwise great start!

File: workflow/lib/preprocess/preprocess.py
Comment (roshankern) [reply workflow/lib/preprocess/preprocess.py:82]: By format are you referring to the single ND2 file with one field of view and multiple channels? I'm not sure if we should assume this format and ignore the potential to use well-based images. Let's discuss over Zoom before we come to a conclusion on this.

File: workflow/lib/shared/illumination_correction.py
Comment (roshankern) [reply workflow/lib/shared/illumination_correction.py:1]: > Might it make sense to have some sort of descriptor for each file? I think this was useful in the original codebase (main) that I had, but up to you.

Are you referring to the docstring at the top of each python file? I think you are right that this would be helpful (I think only really helpful in the lib files). I'll add this to the lib files.

File: workflow/scripts/preprocess/nd2_to_tif.py
Comment (roshankern) [reply workflow/scripts/preprocess/nd2_to_tif.py:11]: Makes sense. I'll check that these are compatible with Fiji/Napari. Definitely don't want to mess up the downstream steps.

File: workflow/lib/shared/illumination_correction.py
Comment (mat10d) [reply workflow/lib/shared/illumination_correction.py:1]: Okay, I think that makes it easier for others to see what each file contains. Yes, referring to the docstring

File: workflow/scripts/preprocess/nd2_to_tif.py
Comment (mat10d) [reply workflow/scripts/preprocess/nd2_to_tif.py:11]: Sounds good

File: workflow/lib/shared/illumination_correction.py
Comment (roshankern) [reply workflow/lib/shared/illumination_correction.py:1]: Done in https://github.com/cheeseman-lab/brieflow/pull/3/commits/654650b7f4e338192d1a6e5f9b3f5947f24c4e9a

File: workflow/scripts/preprocess/nd2_to_tif.py
Comment (roshankern) [reply workflow/scripts/preprocess/nd2_to_tif.py:10]: Confirmed I am able to view these converted tifs. Merging!

**Pull request merged**
